tests/interactor/unit/state_machine/commands/CommandTest.h

```diff
@@ -17,14 +18,17 @@ class CommandTest : public ::testing::Test {
   void SetUp() override {
     auto id_producer = std::make_unique<MockTaskIdProducer>();
     auto task_manager = std::make_unique<TaskManager>(std::move(id_producer));
+    auto persistence = std::make_unique<MockPersistence>();
+    mock_persistence_ = persistence.get();
 
-    model_controller_ =
-        std::make_unique<DefaultModelController>(std::move(task_manager));
+    model_controller_ = std::make_unique<DefaultModelController>(
+        std::move(task_manager), std::move(persistence));
   }
 
  protected:
   std::unique_ptr<ModelController> model_controller_;
   TaskFactory task_factory_;
+  MockPersistence* mock_persistence_;
```

Comment:
better to save reference and use it in EXPECT_CALL?

Reply:
Cannot be done, because reference must be initialized in Initializer list. So, I must use pointer here. Same with `std::reference_wrapper`, it doesnt have default constructor.